src/lib/common/sol-bus.c

```diff
@@ -557,10 +557,18 @@ sol_bus_map_cached_properties(struct sol_bus_client *client,
     SOL_NULL_CHECK(t, -ENOMEM);
 
     t->iface = strdup(iface);
-    SOL_NULL_CHECK_GOTO(t->iface, fail);
+    if (SOL_UNLIKELY(!t->iface)) {
+        r = -errno;
+        SOL_WRN("t->iface == NULL");
```

Comment:
nah, just set "r = -errno" after `strdup()` and before existing `SOL_NULL_CHECK_GOTO()`